jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildTarMojo.java

```diff
@@ -0,0 +1,129 @@
+    ImageReference baseImage = parseImageReference(getBaseImage(), "from");
+    ImageReference targetImage =
+        mavenProjectProperties.getDockerTag(getTargetImage(), mavenBuildLogger);
+
+    // Checks Maven settings for registry credentials.
+    MavenSettingsServerCredentials mavenSettingsServerCredentials =
+        new MavenSettingsServerCredentials(Preconditions.checkNotNull(session).getSettings());
+    RegistryCredentials knownBaseRegistryCredentials =
+        mavenSettingsServerCredentials.retrieve(baseImage.getRegistry());
+
+    String mainClass = mavenProjectProperties.getMainClass(this);
+
+    // Builds the BuildConfiguration.
+    // TODO: Consolidate with BuildImageMojo.
+    BuildConfiguration.Builder buildConfigurationBuilder =
+        BuildConfiguration.builder(mavenBuildLogger)
+            .setBaseImage(baseImage)
+            .setBaseImageCredentialHelperName(getBaseImageCredentialHelperName())
+            .setKnownBaseRegistryCredentials(knownBaseRegistryCredentials)
+            .setTargetImage(targetImage)
+            .setMainClass(mainClass)
+            .setJavaArguments(getArgs())
+            .setJvmFlags(getJvmFlags())
+            .setEnvironment(getEnvironment())
+            .setExposedPorts(ExposedPortsParser.parse(getExposedPorts()))
+            .setAllowHttp(getAllowInsecureRegistries());
+    if (getExtraDirectory() != null && Files.exists(getExtraDirectory())) {
+      try (Stream<Path> extraFilesLayerDirectoryFiles = Files.list(getExtraDirectory())) {
+        buildConfigurationBuilder.setExtraFilesLayerConfiguration(
+            LayerConfiguration.builder()
+                .addEntry(extraFilesLayerDirectoryFiles.collect(Collectors.toList()), "/")
+                .build());
+
+      } catch (IOException ex) {
+        throw new MojoExecutionException(
+            "Failed to list directory for extra files: " + getExtraDirectory(), ex);
+      }
+    }
+    CacheConfiguration applicationLayersCacheConfiguration =
+        CacheConfiguration.forPath(mavenProjectProperties.getCacheDirectory());
+    buildConfigurationBuilder.setApplicationLayersCacheConfiguration(
+        applicationLayersCacheConfiguration);
+    if (getUseOnlyProjectCache()) {
+      buildConfigurationBuilder.setBaseImageLayersCacheConfiguration(
+          applicationLayersCacheConfiguration);
+    }
+    BuildConfiguration buildConfiguration = buildConfigurationBuilder.build();
+
+    // TODO: Instead of disabling logging, have authentication credentials be provided
+    MavenBuildLogger.disableHttpLogging();
+
+    RegistryClient.setUserAgentSuffix(USER_AGENT_SUFFIX);
+
+    try {
+      BuildStepsRunner.forBuildTar(
+              Paths.get(getProject().getBuild().getDirectory()).resolve("jib.tar"),
+              buildConfiguration,
+              mavenProjectProperties.getSourceFilesConfiguration())
+          .build(HELPFUL_SUGGESTIONS);
+      getLog().info("");
```

Comment:
It feels like we should be calling `MavenProjectHelper#attachArtifact()` here?

Reply:
I'm not too familiar with this, but I don't think we'd want to be attaching it as an artifact. I think if a user really wanted it as part of their build artifacts, they can use something like the [build-helper-maven-plugin](https://www.mojohaus.org/build-helper-maven-plugin/attach-artifact-mojo.html).